cachi2/core/package_managers/bundler/parser.py

```diff
@@ -153,7 +154,11 @@ def download_to(self, deps_dir: RootedPath) -> None:
             to_path=git_repo_path.path,
             env={"GIT_TERMINAL_PROMPT": "0"},
         )
-        repo.git.checkout(self.ref)
```

Comment:
If there is no branch specified we might actually want to checkout the ref, don't we? It is either us or bundler during install phase (and that could cascade in a very wrong direction).

Reply:
Yes, but I changed it to `reset --hard <ref>` so we don't end up in the detached state

Reply:
What do you mean by
> it is unlikely that these repos will be later used for development

---
> Since it does not make sense to specify a ref when someone has specified a branch, but they still decided to specify the ref, let's do what they have asked and reset to it anyway.

What if the ref is from a different branch than the specified branch? :smile: Maybe this is an unrealistic scenario, but who knows with bundler?

Reply:
> it is unlikely that these repos will be later used for development

Being in headless state is a problem if someone is to try and commit something to this repository. If they are to build it then it is probably fine (assuming that this ref does not point to a state in which it is unbuildable -- which we cannot protect against and also should not).

> What if the ref is from a different branch than the specified branch? 

Possible, but I'd argue that here we are in the area of deliberate and methodical attempts to shoot oneself in the foot. We cannot shield a user from all possible issues with their repo. We might issue a warning though.

Reply:
But that's not their repo, it is a dependency they specified in the `Gemfile`.

Reply:
> But that's not their repo

Exactly! That's why it is not a problem. I think no change to your current code is needed.